httpserver/service_node.cpp

```diff
@@ -87,18 +135,29 @@ ServiceNode::~ServiceNode() = default;
 
 void ServiceNode::relay_one(const message_t& msg, sn_record_t sn) const {
 
-    BOOST_LOG_TRIVIAL(debug) << "Relaying a message to " << sn;
+    BOOST_LOG_TRIVIAL(debug) << "Relaying a message to: " << sn;
 
     request_t req;
     serialize_message(req.body(), msg);
 
     req.target("/v1/swarms/push");
 
     make_http_request(
-        ioc_, sn.address, sn.port, req, [this, sn](sn_response_t&& res) {
+        ioc_, sn.address, sn.port, req,
+        [this, sn, req = std::move(req)](sn_response_t&& res) {
```

Comment:
`req` seems to be moved inside the lambda by the capture list, then passed by reference to `make_http_request`...

Reply:
Is that not OK?

Reply:
so `make_http_request` gets a req in an undefined state right?

Reply:
Not sure to understand where you're going haha

> It is captured, so its lifetime is until the callback executes

Yes, the lambda captures `req` by move semantic, so now the actual value of req lives only within the lambda's scope. So the callback owns a `req` with valid data, no problems.

> That callback then copies req.body() further.

My point was that the initial variable `req` defined at line 136 is now in an undefined state when passed by reference to `make_http_request`, which passes it on to the `HttpClientSession` constructor, which will try to copy `req.body()`.

To illustrate with a simplified version of the issue:
```cpp
void make_http_request(const std::string& text, std::function<void()> callback) {
    std::cout << text << std::endl; // prints nothing
    callback();  // prints "hello"
}
std::string data = "hello";
make_http_request(data, [ data = std::move(data)] { std::cout << data << std:: endl;});
```

Reply:
Yeah, I see the problem now.

Reply:
Replaced with a copy.